llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -1329,6 +1329,28 @@ SDValue DAGCombiner::reassociateReduction(unsigned RedOpc, unsigned Opc,
                        DAG.getNode(Opc, DL, N0.getOperand(0).getValueType(),
                                    N0.getOperand(0), N1.getOperand(0)));
   }
+
+  // Reassociate op(op(vecreduce(a), b), op(vecreduce(c), d)) into
+  // op(vecreduce(op(a, c)), op(b, d)), to combine the reductions into a
+  // single node.
+  SDValue A, B, C, D;
+  if (sd_match(N0,
+               m_OneUse(m_c_BinOp(Opc, m_OneUse(m_UnaryOp(RedOpc, m_Value(A))),
```

Comment:
Do we need to worry about the reductions having different flags on the operations? Are they expected to match the flags passed in?

Reply:
Thanks for this! Is it worth adding an extra test in CodeGen/AArch64/double_reduct.ll, i.e. something like `nested_mul_f32_mixed_reassoc`?